service/evaluation/evaluation.go

```diff
@@ -0,0 +1,783 @@
+	authz service.AuthorizationStrategy
+}
+
+func init() {
+	log = logrus.WithField("component", "evaluation")
+}
+
+// ServiceOption is a function-style option to configure the Evaluation Service
+type ServiceOption func(*Service)
+
+// WithStorage is an option to set the storage. If not set, NewService will use inmemory storage.
+func WithStorage(storage persistence.Storage) service.Option[Service] {
+	return func(s *Service) {
+		s.storage = storage
+	}
+}
+
+// WithOAuth2Authorizer is an option to use an OAuth 2.0 authorizer
+func WithOAuth2Authorizer(config *clientcredentials.Config) service.Option[Service] {
+	return func(s *Service) {
+		s.SetAuthorizer(api.NewOAuthAuthorizerFromClientCredentials(config))
+	}
+}
+
+// WithAuthorizer is an option to use a pre-created authorizer
+func WithAuthorizer(auth api.Authorizer) service.Option[Service] {
+	return func(s *Service) {
+		s.SetAuthorizer(auth)
+	}
+}
+
+// WithOrchestratorAddress is an option to configure the orchestrator service gRPC address.
+func WithOrchestratorAddress(address string, opts ...grpc.DialOption) service.Option[Service] {
+	return func(s *Service) {
+		s.orchestratorAddress = grpcTarget{
+			target: address,
+			opts:   opts,
+		}
+	}
+}
+
+// NewService creates a new Evaluation service
+func NewService(opts ...service.Option[Service]) *Service {
+	var err error
+	s := Service{
+		orchestratorAddress: grpcTarget{
+			target: DefaultOrchestratorAddress,
+		},
+		scheduler: make(map[string]*gocron.Scheduler),
+		wg:        make(map[string]*sync.WaitGroup),
+	}
+
+	// Apply service options
+	for _, o := range opts {
+		o(&s)
+	}
+
+	// Default to an in-memory storage, if nothing was explicitly set
+	if s.storage == nil {
+		s.storage, err = inmemory.NewStorage()
+		if err != nil {
+			log.Errorf("Could not initialize the storage: %v", err)
+		}
+	}
+
+	// Default to an allow-all authorization strategy
+	if s.authz == nil {
+		s.authz = &service.AuthorizationStrategyAllowAll{}
+	}
+
+	return &s
+}
+
+// SetAuthorizer implements UsesAuthorizer
+func (s *Service) SetAuthorizer(auth api.Authorizer) {
+	s.authorizer = auth
+}
+
+// Authorizer implements UsesAuthorizer
+func (s *Service) Authorizer() api.Authorizer {
+	return s.authorizer
+}
+
+// StartEvaluation is a method implementation of the evaluation interface: It periodically starts the evaluation of a cloud service and the given controls_in_scope (e.g., EUCS OPS-13, EUCS OPS-13.2) in the target_of_evaluation. If no inteval time is given, the default value is used.
+func (s *Service) StartEvaluation(_ context.Context, req *evaluation.StartEvaluationRequest) (resp *evaluation.StartEvaluationResponse, err error) {
+	var (
+		parentSchedulerTag string
+		interval           int
+		toe                *orchestrator.TargetOfEvaluation
+		toeTag             string
+	)
+
+	// Validate request
+	err = service.ValidateRequest(req)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "%v", err)
```

Comment:
You should directly return the `err` here, it's already wrapped in a gRPC error. That is even documented in the function ;)